transformer_engine/pytorch/cpp_extensions/gemm.py

```diff
@@ -54,6 +54,8 @@ def fp8_gemm(
             dtype=out_dtype,
             device="cuda",
         )
+    else:
+        out = out.contiguous()
```

Comment:
We could run into an edge case if the user provides expects this function to output to a non-contiguous buffer. My preferred solution is to just error out in this case:
```suggestion
    if not out.is_contiguous():
        raise ValueError("Output tensor is not contiguous")
```
Alternatively:
```python
    if not out.is_contiguous():
        out.data = out.contiguous()
```
This will not work if `out` is a detached tensor or a view.

Reply:
Suggested change LGTM, instead of changing the `.data`

Reply:
Hmmm, would that be a problem though? Our actual API does not allow users to provide a preallocated output anyway and this would actually prevent noncontiguous inputs from working (at least without other changes) since `out` would be initialized as e.g. `empty_like(input)` and so would be noncontiguous as well.

Reply:
I think having straightforward behavior makes these functions easier to maintain and reason about. They're also used in the operation-based API and maybe by some users.

I don't see any places in the module-based API where the GEMM output is initialized with `empty_like`. It's either [allocated with `empty`](https://github.com/NVIDIA/TransformerEngine/blob/71124c31fbf775d7e5355d7652ddf67e831b7095/transformer_engine/pytorch/module/linear.py#L213), [provided from UB](https://github.com/NVIDIA/TransformerEngine/blob/71124c31fbf775d7e5355d7652ddf67e831b7095/transformer_engine/pytorch/module/linear.py#L189), or [set as a `main_grad`](https://github.com/NVIDIA/TransformerEngine/blob/71124c31fbf775d7e5355d7652ddf67e831b7095/transformer_engine/pytorch/module/linear.py#L556). [The operation-based API also initializes with `empty`](https://github.com/NVIDIA/TransformerEngine/blob/71124c31fbf775d7e5355d7652ddf67e831b7095/transformer_engine/pytorch/ops/basic/basic_linear.py#L477-L502). If it is an issue, we could pass `memory_format=torch.contiguous_format` to `empty_like`.

Reply:
Well, the issue was mostly on the LayerNorm side actually. I changed it according to the suggestion.